src/modules/dashboard/components/HookedTokenIcon/HookedTokenIcon.tsx

```diff
@@ -60,12 +71,22 @@ const HookedTokenIcon = ({
 
   useEffect(() => {
     const loadTokenLogo = async () => {
+      const image = localStorage.getItem(address);
+      if (image) {
+        setTokenImage(image);
+        return;
+      }
+
       if (!dontFetch && address && !iconName) {
         const response = await loadTokenImages(address);
-        if (!response.ok) {
-          return;
+        if (response.ok) {
+          const blob = await response.blob();
+          const base64image = await getBase64image(blob);
+          if (base64image) {
+            localStorage.setItem(address, base64image);
```

Comment:
This sets the base64 string an address as a key value pair in the local storage. This has the potential of polluting the local storage way too much.

Please refactor this to store in under an object.

Something like:
```suggestion
const tokenImages = localStorage.getItem('tokenIcons`);
localStorage.setItem('tokenIcons', { ...tokenImages, [address]: base64image });
```

Then access it like that.